### DISCUSSION TITLE
Unstable Next 14 Migration: Error: Invariant: Expected to replace at least one import

### DISCUSSION THREAD

**Discussion by i-am-mani** [vercel/next.js#63078] [App Router]
TLDR; Unable to work with nextjs-14 project locally after migrating to app dir from next-12, keeps running into `Invariant import` exception, however build and deployments works fine. The error is easily reproducible on our local machines. If anyone could share their experience on similar problem or help me decrypt the error message, i will be very grateful!

Our team has been in a process of migrating from next-12 to latest next-14 release with app dir (no pages dir) and unfortunately we have hit a wall in the process and are unable to make any progress.

Nextjs keeps throwing `Error: Invariant: Expected to replace at least one import` on catch-all segment routes like `[...path]` and we have been unsuccessful in isolating the cause of this problem.

The full error message:

![unstable-next-14-migration-error-invariant-expected-to-v0-a16vtmmqtvlc1](https://github.com/vercel/next.js/assets/45026575/62847e7c-4c74-46e4-930e-f3c11e452c63)
difficult to create a reproduction.

```
./node\_modules/next/dist/build/webpack/loaders/next-route-loader/index.js?kind=PAGES&page=%2F\_error&preferredRegion=&absolutePagePath=.%2Fnode\_modules%2Fnext%2Fdist%2Fpages%2F\_error.js&absoluteAppPath=next%2Fdist%2Fpages%2F\_app&absoluteDocumentPath=next%2Fdist%2Fpages%2F\_document&middlewareConfigBase64=e30%3D!
Error: Invariant: Expected to replace at least one import
```

The error logs doesn't point to any specific file or full stack trace, either in server console logs and runtime red screen. However, surprisingly, the local build as well as preview deployments work fine. 

I am on, "next": "^14.1.0", and in a monorepo project setup with turbo and complete app-dir without `/pages` directory . **Unfortunately, i am unable to create a reproduction on a fresh nextjs project.** The project is huge and without proper error messages its 

I have tried all general workarounds like:
- removing `node_modules` from all projects
- removing `package-lock.json`, building from scratch. No noticeable change was found. 
- removing all dynamic imports

I attempted to take a deep dive into the code execution through debugger, i managed to trace the exception to a file called `load-entrypoint.js`, it's precisely at this file the program throws the exception.

![unstable-next-14-migration-error-invariant-expected-to-v0-kx1h8wcptvlc1](https://github.com/vercel/next.js/assets/45026575/af63b303-fd50-4ccb-b52a-7f48a0232779)

**Comment by Ebrahim-Ramadan** [comment]
sounds like [this issue](https://github.com/vercel/next.js/issues/75825), did they bother to solve this? or what should I do

also [found this](https://nextjs-forum.com/post/1295402866135269447#message-1295404666984595567)

**Reply by ViniciusCestarii** [reply]
Thanks @Ebrahim-Ramadan! The last link you shared was the solution for my problem:

Invariant: Expected to replace all template variables, missing VAR_ORIGINAL_PATHNAME in template

I just deleted .next folder and it worked

**Comment by Olwiba** [comment]
I was running into a similar problem. Check your node version and make sure it's supported. Upgrade if necessary :)

**Comment by mado-opus** [comment]
Try redeploying your branch on Vercel without "Use existing build cache".
From https://vercel.com/docs/deployments/troubleshoot-a-build#understanding-build-cache:

> If you create a new Git branch in MyCode and make a commit to it, there is no cache for that specific branch. In this case, the last [production Deployment](https://vercel.com/docs/deployments/environments#production) cache is used to create a preview deployment and a new branch cache is created for subsequent commits to the new branch.

**Comment by bgw** [comment]
Closing:
- This issue is likely fixed in newer versions of Next.js. Please open a new issue if it is not.
- This sort of thing can sometimes happen if you're running a different version of next.js than the one installed in your repository. See my comment here: https://github.com/vercel/next.js/discussions/89081#discussioncomment-15676666
